edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/logs/LogsProvider.cs

```diff
@@ -21,16 +23,65 @@ public LogsProvider(IRuntimeInfoProvider runtimeInfoProvider, ILogsProcessor log
 
         public async Task<byte[]> GetLogs(ModuleLogOptions logOptions, CancellationToken cancellationToken)
         {
+            Preconditions.CheckNotNull(logOptions, nameof(logOptions));
             Stream logsStream = await this.runtimeInfoProvider.GetModuleLogs(logOptions.Id, false, Option.None<int>(), cancellationToken);
+            Events.ReceivedStream(logOptions.Id);
+
             byte[] logBytes = await this.GetProcessedLogs(logsStream, logOptions);
             return logBytes;
         }
 
+        public async Task GetLogsStream(ModuleLogOptions logOptions, Func<ArraySegment<byte>, Task> callback, CancellationToken cancellationToken)
```

Comment:
What is the callback used for?

Reply:
The callback is invoked on each frame in the incoming stream. That way the method does not return anything, simply processes the stream, with the callback expected to include the final processing logic. 
I wanted to call it ProcessLogsStream, but I thought it would get confusing since I already have a LogsProcessor that "processes" the logs. Any other recommendations for the name?

Reply:
This will help when we need to Process the incoming stream (like filter, compress, etc). before sending it over the ws